src/rmc-compiler/rust-toolchain.toml

```diff
@@ -0,0 +1,6 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+[toolchain]
+channel = "nightly-2021-12-27"
```

Comment:
Does this cause cargo to download this toolchain, or does it fail or something if it's not present?

Reply:
It will download it.

Reply:
I believe any rustup command will trigger the download

Reply:
The plan is to update this every two weeks for now. Instead of merging with the upstream repository, we will keep up with the latest compiler by updating the toolchain file.

I don't know long they plan to keep old toolchains, but from what I've seen, they have been keeping them all on S3 and some linux distro uses old versions of rustup. For documentation purpose, we could also track the commit that is associated with the current version.

Unfortunately, we still rely on a lot of nightly features, so we can't just use stable releases.